File: src/main/java/land/leets/domain/interview/usecase/GetInterviewImpl.java
package land.leets.domain.interview.usecase;

import land.leets.domain.application.domain.Application;
import land.leets.domain.interview.domain.Interview;
import land.leets.domain.interview.domain.repository.InterviewRepository;
import land.leets.domain.interview.presentation.dto.res.InterviewResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.Optional;

@Service
@RequiredArgsConstructor
public class GetInterviewImpl implements GetInterview {

    private final InterviewRepository interviewRepository;

    @Override
    public InterviewResponse execute(Application application) {
        Optional<Interview> interview = interviewRepository.findByApplication(application);
        return interview.map(InterviewResponse::from).orElse(null);
    }
}

File: src/main/kotlin/land/leets/domain/interview/usecase/GetInterviewImpl.kt
package land.leets.domain.interview.usecase

import land.leets.domain.application.domain.Application
import land.leets.domain.interview.domain.repository.InterviewRepository
import land.leets.domain.interview.presentation.dto.res.InterviewResponse
import org.springframework.stereotype.Service

@Service
class GetInterviewImpl(
    private val interviewRepository: InterviewRepository
) : GetInterview {

    override fun execute(application: Application): InterviewResponse {
        val interview = interviewRepository.findByApplication(application)
        return interview.map { InterviewResponse.from(it) }
            .orElse(InterviewResponse(null, null))
    }
}
